File: core/src/main/java/com/lomekwi/cave/ui/widget/EllipsisLabel.java
package com.lomekwi.cave.ui.widget;

import com.kotcrab.vis.ui.widget.Tooltip;
import com.kotcrab.vis.ui.widget.VisLabel;

public class EllipsisLabel extends VisLabel {
    public EllipsisLabel(String text, int maxChars) {
        super(truncate(text, maxChars));
        new Tooltip.Builder(text).target(this).build();
    }

    private static String truncate(String text, int maxChars) {
        if (text == null) return "";
        if (text.length() <= maxChars) return text;
        return text.substring(0, maxChars) + "...";
    }
}

File: core/src/main/scala/com/lomekwi/cave/ui/widget/EllipsisLabel.scala
package com.lomekwi.cave.ui.widget

import com.kotcrab.vis.ui.widget.Tooltip
import com.kotcrab.vis.ui.widget.VisLabel

class EllipsisLabel(text: String, maxChars: Int) extends VisLabel(EllipsisLabel.truncate(text, maxChars)) {
  new Tooltip.Builder(text).target(this).build()
}

object EllipsisLabel {
  private def truncate(text: String, maxChars: Int): String = {
    if (text == null) ""
    else if (text.length() <= maxChars) text
    else text.substring(0, maxChars) + "..."
  }
}
